lib/gpu/render_pass.cc

```diff
@@ -360,20 +345,25 @@ void InternalFlutterGpu_RenderPass_BindVertexBufferHost(
                    view->range.length, vertex_count);
 }
 
-template <typename TBuffer>
-static void BindIndexBuffer(flutter::gpu::RenderPass* wrapper,
-                            TBuffer buffer,
-                            int offset_in_bytes,
-                            int length_in_bytes,
-                            int index_type,
-                            int index_count) {
-  auto& vertex_buffer = wrapper->GetVertexBuffer();
-  vertex_buffer.index_buffer = impeller::BufferView{
+static void BindIndexBuffer(
+    flutter::gpu::RenderPass* wrapper,
+    const std::shared_ptr<const impeller::DeviceBuffer>& buffer,
+    int offset_in_bytes,
```

Comment:
It may not matter if these are internal APIs but using signed ints for values that can't be negative is scary :)

Reply:
Is there an alternative approach we should consider exploring with Dart FFI?

Reply:
dart::ffi methods support signed/unsigned types, and then you validate on the dart side of things. Or else check on the C++ and then cast.

Reply:
Ah yeah we should use those unsigned types, especially for enums. Added an issue here: https://github.com/flutter/flutter/issues/157129